RoddyCore/test/src/de/dkfz/roddy/config/validation/DefaultValidatorSpec.groovy

```diff
@@ -0,0 +1,97 @@
+    }
+
+    def 'test check proper dollar sign usage in variables'(value, type, expected, expectedWarnings, expectedErrors) {
+        when:
+        ConfigurationValue cval = new ConfigurationValue('bla', value, type)
+
+        then:
+        cval.warnings == expectedWarnings
+        cval.errors == expectedErrors
+        cval.valid == expected
+
+        where:
+        value                   | type     | expected | expectedWarnings                          | expectedErrors
+        // All are ok
+        'abc${var}bcd${var}'    | 'string' | true     | []                                        | []
+        'abc${var}bcd${var}def' | 'string' | true     | []                                        | []
+
+        '\\$'                   | 'string' | true     | []                                      | []  //Escaped dollar
+        'abc\\$'                | 'string' | true     | []                                      | []  //Escaped dollar
+        'abc\\${'               | 'string' | true     | []                                      | []  //Escaped dollar with open brace
+        'a\\${b}c'              | 'string' | true     | []                                      | []  //Escaped dollar with complete braces
+
+        // Not ok
+        'abc$'                  | 'string' | false    | [detachedDollarCharacter.expand('bla')] | []
+        'tr$ue'                 | 'string' | false    | [detachedDollarCharacter.expand('bla')] | []
+        '$'                     | 'string' | false    | [detachedDollarCharacter.expand('bla')] | []
+        'abc$\\{'               | 'string' | false    | [detachedDollarCharacter.expand('bla')] | []
+    }
+
+    def "test check for proper variable usage definition"() {
+        when:
+        ConfigurationValue cval = new ConfigurationValue('bla', value, type)
+
+        then:
+        cval.warnings == expectedWarnings
+        cval.errors == expectedErrors
+        cval.valid == expected
+
+        where:
+        value              | type     | expected | expectedWarnings                           | expectedErrors
+        'abc${var}'        | "string" | true     | []                                         | []
+        'abc\\${var}'      | "string" | true     | []                                         | []
+        'abc${var}v${var}' | "string" | true     | []                                         | []
+        '\\${c}'           | 'string' | true     | []                                         | []
+        '\\${c'            | 'string' | true     | []                                         | []
+        'abc${${var}}'     | "string" | false    | [inproperVariableExpression.expand('bla')] | []
+        'abc${'            | 'string' | false    | [inproperVariableExpression.expand('bla')] | []
+        'ab${c'            | 'string' | false    | [inproperVariableExpression.expand('bla')] | []
```

Comment:
"${}" is still missing. One may think that this got less likely, now that nested variables "${${x}}" are forbidden, but be aware that the whole mechanism is just like Bash's. What about the following situation:
```
a='\${'
b='}'
c='${a}${b}'='${}'
```
Or would this not be possible, because the outcome would be `'\${}'`, i.e. with an escaped dollar?

I did not check in detail, but maybe more examples I suggested may be missing.

Reply:
If Roddy resolves c, then c would like like: '\${}' at the end.